src/HydrationMiddleware/UpdateHistoryPath.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Livewire\HydrationMiddleware;
+
+use Illuminate\Support\Str;
+use Livewire\Exceptions\CannotCombineHistoryPathAndQueryStringException;
+use Livewire\Exceptions\InvalidHistoryPathException;
+
+class UpdateHistoryPath implements HydrationMiddleware
+{
+    public static function hydrate($instance, $request)
+    {
+        //
+    }
+
+    public static function dehydrate($instance, $response)
+    {
+        if (! empty($url = $instance->mapStateToUrl())) {
+        	$root = url()->to('/');
+	        
+        	throw_if(isset($response->updatesQueryString), new CannotCombineHistoryPathAndQueryStringException($instance->getName()));
```

Comment:
Theoretically, we could merge both the query string and the path, but it's a lot more legwork for a use-case that is unlikely (you'll probably use one or the other).